### PR TITLE
Fix / refactor parameter mapping (breaking change)

### LINKED ISSUE #342: Incorrect scale for log10 parameters with estimate=False?
Following parameter definition:

parameterId | parameterScale | lowerBound | upperBound | estimate | nominalValue
-- | -- | -- | -- | -- | --
lambda_reg | log10 | 0.001 | 1000.0 | FALSE | 10.0

In

https://github.com/PEtab-dev/PEtab/blob/8b03c8f20c8042c5375adabfe23bab19c3a649b0/petab/parameter_mapping.py#L349

`parameters.scale` rescales the nominal value to logarithmic coordinates and sets it to a numeric value.

https://github.com/PEtab-dev/PEtab/blob/8b03c8f20c8042c5375adabfe23bab19c3a649b0/petab/parameter_mapping.py#L446

only sees a numeric value and claims a linear scale, which means that the value will be passed to the simulator on an incorrect scale, right?

I feel like this is very basic functionality and should be covered by tests.

### PR DESCRIPTION
**Pull request by dweindl**
Fixes #342

Parameter value and parameter scale mapping now always done together.
Fixed handling of parameter scale for non-estimated scaled parameters.

### CODE REVIEW

File: tests/test_parameter_mapping.py [hunk 1]
```diff
@@ -139,28 +165,45 @@ def test_all_override(condition_df_2_conditions,
...
+                 'observableParameter2_obs1': 'obs1par2cond2override',
+                 'observableParameter1_obs2': 'obs2par1cond2override'
+                 },
+                {},
+                {'dynamicParameter1': LIN, 'dynamicParameter2': LIN,
+                 'observableParameter1_obs1': LIN,
+                 'observableParameter2_obs1': LIN,
+                 'observableParameter1_obs2': LIN
+                 }
+             )
```
Comment (dweindl) [inline tests/test_parameter_mapping.py]: Codacy Issue found: [Wrong hanging indentation.](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5074051)

File: tests/test_parameter_mapping.py [hunk 2]
```diff
@@ -139,28 +165,45 @@ def test_all_override(condition_df_2_conditions,
...
+                 'observableParameter2_obs1': 'obs1par2cond1override',
+                 'observableParameter1_obs2': 'obs2par1cond1override',
+                 },
+                {},
+                {'dynamicParameter1': LIN, 'dynamicParameter2': LIN,
+                 'observableParameter1_obs1': LIN,
+                 'observableParameter2_obs1': LIN,
+                 'observableParameter1_obs2': LIN
+                 }
+             ),
```
Comment (dweindl) [inline tests/test_parameter_mapping.py]: Codacy Issue found: [Wrong hanging indentation.](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5074051)

File: tests/test_parameter_mapping.py [hunk 3]
```diff
@@ -90,12 +107,21 @@ def test_no_condition_specific(condition_df_2_conditions,
                      {'dynamicParameter1': np.log(11.0),
                       'dynamicParameter2': 'dynamicParameter2',
                       'dynamicParameter3': 'dynamicParameter3',
-                      'fixedParameter1': 1.0}),
+                      'fixedParameter1': 1.0},
+                    {},
```
Comment (dweindl) [inline tests/test_parameter_mapping.py]: Codacy Issue found: [Wrong continued indentation (add 1 space).](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5074051)

File: tests/test_parameter_mapping.py [hunk 4]
```diff
@@ -90,12 +107,21 @@ def test_no_condition_specific(condition_df_2_conditions,
...
+                    {'dynamicParameter1': LOG,
```
Comment (dweindl) [inline tests/test_parameter_mapping.py]: Codacy Issue found: [Wrong continued indentation (add 1 space).](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5074051)

File: petab/parameter_mapping.py [hunk 5]
```diff
@@ -58,12 +60,22 @@ def get_optimization_to_simulation_parameter_mapping(
...
 
     Returns:
-        The length of the returned array is n_conditions, each entry is a tuple
-        of two dicts of length n_par_sim, listing the optimization parameters
-        or constants to be mapped to the simulation parameters, first for
-        preequilibration (empty if no preequilibration condition is specified),
-        second for simulation. ``NaN`` is used where no mapping exists.
+        Parameter value and parameter scale mapping for all conditions.
+
+        The length of the returned array is the number of unique combinations
```
Comment (yannikschaelte) [inline petab/parameter_mapping.py]:
```suggestion
        The lengths of the returned array are respectively the numbers of unique combinations
```

File: petab/parameter_mapping.py [hunk 6]
```diff
@@ -58,12 +60,22 @@ def get_optimization_to_simulation_parameter_mapping(
...
+        of simulationConditionId and preequilibrationConditionId from the
```
Comment (yannikschaelte) [inline petab/parameter_mapping.py]:
```suggestion
        of simulationConditionIds and preequilibrationConditionIds from the
```

File: tests/test_parameter_mapping.py [hunk 7]
```diff
@@ -39,19 +39,27 @@ def test_no_condition_specific(condition_df_2_conditions,
                      {'dynamicParameter1': 1.0,
                       'dynamicParameter2': 2.0,
                       'dynamicParameter3': 3.0,
-                      'fixedParameter1': 1.0}),
+                      'fixedParameter1': 1.0},
+                    {},
```
Comment (dweindl) [inline tests/test_parameter_mapping.py]: Codacy Issue found: [Wrong continued indentation (add 1 space).](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5074051)

File: petab/parameter_mapping.py [hunk 8]
```diff
@@ -342,122 +360,50 @@ def _apply_parameter_table(mapping: ParMappingDict,
...
+            else:
+                scale_mapping[row.Index] = LIN
+            par_mapping[row.Index] = val
         else:
-            mapping[row.Index] = row.Index
+            par_mapping[row.Index] = row.Index
 
     # Replace any leftover mapped parameter coming from condition table
-    for key, value in mapping.items():
+    for key, value in par_mapping.items():
```
Comment (yannikschaelte) [inline petab/parameter_mapping.py]: use more descriptive than key and value?

File: tests/test_parameter_mapping.py [hunk 9]
```diff
@@ -314,88 +375,98 @@ def test_parameterized_condition_table_changed_scale(
...
-        assert actual_par_map == expected_par_map
-        assert actual_scale_map == expected_scale_map
+        # test without preequilibration condition; scaled known parameters
+
+        actual = petab.get_optimization_to_simulation_parameter_mapping(
+            measurement_df=measurement_df,
+            condition_df=condition_df,
+            parameter_df=parameter_df,
+            sbml_model=model,
+            scaled_parameters=True
```
Comment (yannikschaelte) [inline tests/test_parameter_mapping.py]: good

File: petab/parameter_mapping.py (hunk 5 above)
Comment (dweindl) [reply petab/parameter_mapping.py]: I'd say it was correct before. It's an array of tuples, not a tuple of arrays.

File: tests/test_parameter_mapping.py (hunk 7 above)
Comment (dweindl) [inline tests/test_parameter_mapping.py]: Codacy Issue found: [Wrong continued indentation (add 1 space).](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=5074051)

File: petab/parameter_mapping.py (hunk 5 above)
Comment (yannikschaelte) [reply petab/parameter_mapping.py]: of course, you are right!

**Pull request merged**